src/Application/Queries/GetPaginatedRequisitions.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Application.Paging;
+using Application.Repositories;
+using Domain.Enums;
+using Mapster;
+using MediatR;
+
+namespace Application.Queries
+{
+    public class GetPaginatedRequisitions
+    {
+        public record Query : PageRequest, IRequest<PaginatedList<RequisitionResponse>>
+        {
+            public DateTime? StartDate { get; set; }
+            public DateTime? EndDate { get; set; }
+            public string? ExpenseHead { get; set; }
+            public string? Department { get; set; }
+            public string? StatusFilter { get; set; } //Assumption is that FE sends comma separated values
+            public HashSet<int> RequisitionStatuses => string.IsNullOrWhiteSpace(StatusFilter) ? new HashSet<int>() : StatusFilter.Split(',').Select(int.Parse).ToHashSet();
+            public string? TypeFilter { get; set; } //Assumption is that FE sends comma separated values
```

Comment:
again, please remove assumptions in implementation

Reply:
Removed.